### ISSUE TITLE
[Feature] Multiple payload with single sign

### ISSUE THREAD

**Issue by Cadey** [XRPL-Labs/XUMM-Issue-Tracker#425]
**The Problem**
Now that nfts are live on the ledger users are faced with having to sign multiple requests to claim, sell or buy their nfts.

**The solution**
Allow for multiple payloads to be sent in the same request to the xumm Api, each of which can be accepted or rejected in a single validation request.

**Additional concerns**
If too many payloads are sent it could be challenging for the user to validate them all which in turn could lead to scrupulous people trying to hide a transaction within a bunch.

One way to mitigate this could be to force all multiple payload requests to be of the same type, such as create offer, or accept offer. This would allow for a much simpler Ui to be displayed in the xumm app.

**The result**
1) When a multiple payload request is made, a single action is created (qr code) for the user to scan.

2) Once scanned the xumm app will show the user all the transactions in this collection.

3) The xumm app allows the user to view each request and accept/reject each one. Ideally displayed in a list/table layout

4) The user accepts the final list of transactions and the xumm app then signs each one and sends them to the ledger. Excluding and the user rejected.

5) The scan result message sent from xumm will now include multiple results, guaranteed to be in the same order as the payloads sent in. Rejected payloads are included and look the same as a non signed (defined) request does today. 

5) With exception of the scan result, hooks and websocket responses all stay as they are since each transaction on the ledger is still singular.

**Comment by WietseWind** [comment]
This has been something we've discussed before (and much more internally, but an excerpt of those discussions hasn't been updated in the issue, #352). I'll close the previous issue as your report is more clear as to why this request should be discussed. Thank you!

Reasons for getting this request (in the past, and now):
- Airdrops and the likes (don't send with Xumm, build a tool, sign locally, programattically)
- Enterprise batch payments (don't use Xumm, use a specialised enterprise wallet: enterprises need more specialised things anyway)
- XLS20 (NFT's on XRPL) minting

Some of the things we discussed internally involve:
- Creating multiple payloads
- The grouping them in one "batch payload"
- Then offering the end user a card-like (+ progress bar) interface where users have to navigate (prev/next) between all included payloads, and at the end: approve all of them.

**HOWEVER**... I strongly feel we should not implement this, and I will explain why (bear with me 😉). I really believe this is a perfect example of the famous Henry Ford quote: 

> **If I had asked people what they wanted, they would have said faster horses.**

My arguments against adding "multi payload signing" (aka "batch signing") can be divided into two categories:
- Unfeasible from user experience standpoint
- Unfeasible from a tech standpoint

Your suggestion to allow batch signing for _the same transaction types_ is a great approach, but I think there are other things to consider:

# User experience
- If users get a sequence of say 20 payloads, the first one: they will take a good look at. Then a couple of them with less attention. Then users will just next/next/next and sign. This is very risky: an attacker could (and we're in crypto, so definitely: will) sneak in something uncanny in one of the last ones. 
- Unfeasible for users _and_ developers: say the connectivity (or node) is lost/unresponsible halfway signing & submitting: where did it left off? What went right? What didn't went right?  How to explain that to the user? How to explain that to the dev? Many (especially new) developers still wrestle with the basics or even single transaction "reliable transaction submission". How can we expect this to go well if a batch has to be verified, partially re-done, etc.? Also: every transaction needs to have an incremented `Sequence`. Say in a batch of 20 transactions, no. 11 has been crafted incorrectly (or that's where the account hits a restriction): this will make 12-20 fail as well.

# Tech
Back to the Henry Ford quote. If a batch of NFT's are minted: why would that require batch signing in the first place? Instead of creating a somewhat unreliable/unfeasible dev & UI flow, I feel we should address the underlying issue: **the fact that batch signing is even required at all for a minting process**.

### First suggested potential technical implementation

Let's say one wants to mint a bunch of NFT's using a tool/platform:
1. User temporarily configures the tool/platform as `NFTokenMinter` on their account
2. Tool/platform mints
3. Tool/platform creates the preferred offers

Step 2 ~ 3 is where we the proposed batch signing comes in, but if an account already authorized a platform to mint for them, why not (possibly even conditionally) give the platform the permission to create the offers post-minting? Until their permission is revoked?

### Second suggested potential technical implementation

If it would make sense to add batch-transactions globally to the XRPL, it would make much more sense not to make it a client side (wallet UI) & developer responsibility, but an XRPL responsibility. Where **either all transactions, or none of the transactions at all** are to succeed in one batch. A solution would be to add a `ttBATCH` transaction type, containing an array of Transactions (omitting the Sequence) in the regular XRPL transaction format. A client (code, wallet, ...) can then sign the entire batch, at which point the batch is applied to the XRPL like a traditional SQL Transaction: either all of them succeed, or none of them are applied.

# So, TL;DR:
- Users shouldn't be responsible of carefully looking at an entire range of proposed transactions
- Users shouldn't be burdened with half-succeeded signing sequences
- Developers shouldn't be burdened with half-succeeded signing sequences
- A feasible, reliable technical solution can be found and implemented to take away the need for batch signing in the first place (either XLS20 specific, or even XRPL transaction type wide with the `ttBATCH` approach)

My personal preference: the `ttBATCH` implementation.
